corehq/apps/domain/auth.py

```diff
@@ -54,6 +56,13 @@ def determine_authtype_from_header(request, default=DIGEST):
 
 
 def determine_authtype_from_request(request, default=DIGEST):
+    """
+    There's a bug on mobile where demo submissions weren't always
+    sending noauth headers, but that should be the default.
+    """
+    if request.GET.get('submit_mode') == DEMO_SUBMIT_MODE:
+        return NOAUTH
```

Comment:
having two docstrings like this in a function is a bit weird. Think you could move this whole block down below the original docstring, and change yours into a regular comment?
Also, it could be useful to include the specific version where this bug was fixed on mobile.

Reply:
Updated